### ISSUE TITLE
ISSUE: connectionName parameter does not works!

### ISSUE THREAD

**Issue by ytpl65** [KnowageLabs/Knowage-Server#899]
knowage server: 8.1
ubuntu os: 22.04
BIRT-report-designer: 4.13.0

1. Datasource Connection:
![image](https://github.com/KnowageLabs/Knowage-Documentation/assets/89214284/87da53dd-b03b-417f-8225-658372a8d3ac)
2. Dataset Preview
![image](https://github.com/KnowageLabs/Knowage-Documentation/assets/89214284/6c4d5557-79ca-46ee-acf0-5743c9654b35)
3. Report Parameters
![image](https://github.com/KnowageLabs/Knowage-Documentation/assets/89214284/4721e740-90b8-4452-972f-3cbb3f81e233)
4. connectionName parameter:
![image](https://github.com/KnowageLabs/Knowage-Documentation/assets/89214284/20318e04-660d-4d49-a43e-ad347ea3cd32)
5. Parameter Binding
![image](https://github.com/KnowageLabs/Knowage-Documentation/assets/89214284/fd51a7b0-8470-40c4-af38-a038e1e306c6)
6. Birt Report Execution:
![image](https://github.com/KnowageLabs/Knowage-Documentation/assets/89214284/241f8bc6-5153-4c10-a630-c74a1998cddb)
7. Data Source configured at knowage server:
![image](https://github.com/KnowageLabs/Knowage-Documentation/assets/89214284/7e081b21-2839-4d50-a173-662b3f713223)
8. Deployed at Knowage server:
![image](https://github.com/KnowageLabs/Knowage-Documentation/assets/89214284/6a448223-7c62-4f4a-9cf2-8e6650c7193e)
9. Analytical Driver:
![image](https://github.com/KnowageLabs/Knowage-Documentation/assets/89214284/447bf26f-4a43-41f7-ac68-69007a10caa0)
10. Report Executed at knowage server
![image](https://github.com/KnowageLabs/Knowage-Documentation/assets/89214284/8e6b823d-d2f9-4708-9a01-4aeb9dac0972)

Now, Compare the data of screenshots of  No.6 and No. 10. it is same!
The Knowage server is not using the correct data source when running reports. Even though a connection name parameter is passed when executing the report on the server, it is not pulling data from that data source.

Instead, it is always using the data source defined in the original BIRT report design, regardless of the connection name passed.

The behavior should be:

When a report runs on Knowage server and a connection name parameter is provided, it should fetch and display data from that specified data source.

But it is not working that way. It always defaults back to using the data source hardcoded in the BIRT report design, rather than using the one provided in the connection name parameter as it's supposed to.

So in short, the server is not respecting the connection name parameter and using the wrong data source when reports are executed.

Help Me
Thank You
